ioservice/io_foms.c

```diff
@@ -1974,6 +1974,7 @@ static int io_sync(struct m0_fom *fom)
 	struct m0_be_tx         *tx   = m0_fom_tx(fom);
 
 	if ((fobj->fcrw_flags & M0_IO_FLAG_SYNC) &&
+	    fom->fo_tx.tx_state != M0_DTX_INVALID &&
```

Comment:
Can you also add this change?
```
diff --git a/ioservice/io_foms.c b/ioservice/io_foms.c
index 280cb323..5bc9b5e3 100644
--- a/ioservice/io_foms.c
+++ b/ioservice/io_foms.c
@@ -2241,7 +2241,7 @@ static int m0_io_fom_cob_rw_tick(struct m0_fom *fom)
 
        if (m0_is_write_fop(fom->fo_fop) &&
            m0_fom_phase(fom) == M0_FOPH_IO_ZERO_COPY_WAIT &&
-           fom->fo_tx.tx_state == 0) {
+           fom->fo_tx.tx_state == M0_DTX_INVALID) {
                m0_fom_phase_set(fom, M0_FOPH_TXN_INIT);
                return M0_RC(M0_FSO_AGAIN);
        }

```

It is not related to this issue. But I think it is helpful.

Reply:
I surely can. A separate PR for unrelated code change: https://github.com/Seagate/cortx-motr/pull/793.

Reply:
Thank you.